client/src/GameState.ts

```diff
@@ -22,6 +34,52 @@ export class GameState {
     otherTetrominoes: Array<Tetromino>;
     playerId!: 0 | 1 | 2 | 3;
 
+    lineCheckSequence!: Array<LineCheckTask>;
+    // template used to generate lineCheckSequence once playerId is received
+    static LineCheckTemplate: Array<LineCheckTask> = [
+        {
+            axis: LineCheckAxis.Row,
+            direction: 1,
+            range: [],
+        },
+        {
+            axis: LineCheckAxis.Column,
+            direction: -1,
+            range: [],
+        },
+        {
+            axis: LineCheckAxis.Row,
+            direction: -1,
+            range: [],
+        },
+        {
+            axis: LineCheckAxis.Column,
+            direction: 1,
+            range: [],
+        },
+    ];
+
+    private generateLineCheckSequence(
+        playerId: number = 0
+    ): Array<LineCheckTask> {
+        const sequence = GameState.LineCheckTemplate;
+        const distanceToPlayer0 = (4 - playerId) % 4;
+        // rotate the line checking sequence to always start from player 0
+        [...Array(distanceToPlayer0).keys()].forEach((_) => {
+            sequence.push(sequence.shift()!);
+        });
+
+        // for each task, generate the range of lines to check with, i.e. rows from top to center (before rotation)
+        // e.g. for current player it will be [0..19], for opposite it's [39..20]
+        return sequence.map(({ axis, direction }) => {
+            let range = [...Array(BOARD_SIZE / 2).keys()].map(
+                (i) => BOARD_SIZE - 1 + i * direction
+            );
+            if (direction === 1) range = range.map((x) => (x + 1) % BOARD_SIZE);
+            return { axis, direction, range };
+        });
```

Comment:
Imo this not as readable as it could be and could perhaps be simplified with something like below (though I haven't thought too hard about it, maybe not)
```ts
const halfBoard = [...Array(BOARD_SIZE / 2).keys()];
return sequence.map(({ axis, direction }) => {
    if (direction > 0) return { axis, direction, halfBoard.slice() };
    return { axis, direction, halfBoard.map((i) => BOARD_SIZE - 1 - i };
});
```